src/lib/parser.rs

```diff
@@ -108,8 +108,51 @@ impl Parser {
     Ok(parser)
   }
   pub fn next_char(&mut self) -> Option<char> {
+    // get next
     let letter = self.contents.get(self.index)?;
+
+    // define forward slash, newline & astrix
+    let fs: u8 = '/' as u8;
+    let nl: u8 = '\n' as u8;
+    let ast: u8 = '*' as u8;
```

Comment:
I think you don't need the `: u8` here
```suggestion
    let fs = '/' as u8;
    let nl = '\n' as u8;
    let ast = '*' as u8;
```